### PR TITLE
Register

### PR DESCRIPTION
**Pull request by KrzysztofUrbanczyk**
Registration and Login with Name/Pass

### CODE REVIEW

File: Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+            public async Task<(bool Success, string Result)> Handle(RegisterUserCommand request, CancellationToken cancellationToken)
+            {
+                //TODO: move validation somewhere 
+                if (_db.Users.Any(x => x.Name == request.Name))
+                {
+                    return (false, "An account with the given username already exists.");
+                }
+
+                if (_db.Users.Any(x => x.Email == request.Email))
```
Comment (dotNETSanta) [inline Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommand.cs]: What if there's **Email** "example@gmail.com in database and I will insert "Example@gmail.com", same with **Name**.

Should we treat them as the same? Especially user names can be confusing **somekind somekInd**

Also where we can define e.g password complexity rules? we gotta write them by hand here, or?

File: Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommandValidation.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    public class RegisterUserCommandValidation : AbstractValidator<RegisterUserCommand>
+    {
+        public RegisterUserCommandValidation()
+        {
+            RuleFor(x => x.Name).NotEmpty().WithMessage("Name is required.");
+            RuleFor(x => x.Email).NotEmpty().WithMessage("Email is required.")
+                                 .EmailAddress().WithMessage("A valid email is required.");
+            RuleFor(x => x.Password).NotEmpty().WithMessage("Password is required.")
+                                    .MinimumLength(10).WithMessage("The password should be at least 10 characters long.");
+        }
```
Comment (dotNETSanta) [inline Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommandValidation.cs]: I think that here's room for improvement when it comes to readability

File: Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommand.cs (hunk 1 above)
Comment (KrzysztofUrbanczyk) [reply Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommand.cs]: You are right, Name and Email should ignore case, fixed.

By `TODO` I mean to move it to another class to make it cleaner.

Password rules can be in `RegisterUserCommandValidation`.
I don't like complex rules, but we can do it like this:
https://stackoverflow.com/a/45126929

File: Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommandValidation.cs (hunk 2 above)
Comment (KrzysztofUrbanczyk) [reply Coyote/CoyoteNETCore.Application/Account/Commands/RegisterUserCommandValidation.cs]: I like this "fluent" way.

**Pull request merged**